docs/V5-Migration-Guide.md

```diff
@@ -0,0 +1,84 @@
+# Version 5.0.0 Migration Guide
+
+Dojo 5.0.0 includes a few breaking changes that will need to some changes in your applications code base. As with previous releases, we have updated the `@dojo/cli-upgrade-app` CLI command to help you through the upgrade process and automate as much as possible. Where it is not possible to automate the upgrade, you should receive helpful information in the output that indicates what needs to get manually upgraded and in which module.
+
+To install the upgrade command, run the following from the project root:
+
+```
+npm install @dojo/cli-upgrade-app --no-save
+```
+
+To perform the migration, run the following command from the project root. The migrations tool should automatically detect the necessary migration from your `package.json`.
+
+```
+dojo upgrade app
+```
+
+If you are upgrading from a version before 4.0.0, please see the [previous migration guides](./) for more details first.
+
+* The `@dojo/cli` should be updated to version 5.0.0, along with all the commands used by the project.
+* If your project is using @dojo/widgets and @dojo/interop, these packages also require upgrading to version 5.0.0.
+
+**Note:** The migration tool may create line lengths that violate your projects linting rules, be sure to run your linter and manually fix any linting rule violations.
+
+### Breaking Changes:
+
+#### [has consolidation](https://github.com/dojo/framework/pulls)
+
+Part of the legacy clean up, this release of Dojo consolidates all the `has` related modules in a single `has` module, `@dojo/framework/has/has`. If you run the `dojo/cli-upgrade-app` command all references to removed has modules should be automatically updated.
```

Comment:
Will someone know the context for the legacy cleanup? Maybe say something like "Part of the cleanup started with the 3.0 consolidation of packages into `@dojo/framework` ?